src/components/numeral-date/numeral-date.stories.tsx

```diff
@@ -63,6 +65,96 @@ export const InternalValidationWarning: ComponentStory<
   );
 };
 
+export const Validation: ComponentStory<typeof NumeralDate> = () => {
+  const [value, setValue] = useState({ dd: "", mm: "", yyyy: "" });
+  return (
+    <>
+      <NumeralDate
+        label="Validation as string"
+        error="Error Message (Fix is required)"
+        onChange={(e) => setValue(e.target.value)}
+        value={value}
+      />
+
+      <NumeralDate
+        label="Validation as string on label"
+        error="Error Message (Fix is required)"
+        validationOnLabel
+        onChange={(e) => setValue(e.target.value)}
+        value={value}
+      />
+
+      <NumeralDate
+        label="Validation as boolean"
+        error
+        onChange={(e) => setValue(e.target.value)}
+        value={value}
+      />
+
+      <NumeralDate
+        label="Validation as string"
+        warning="Warning Message (Fix is optional)"
+        onChange={(e) => setValue(e.target.value)}
+        value={value}
+      />
+
+      <NumeralDate
+        label="Validation as string on label"
+        warning="Warning Message (Fix is optional)"
+        validationOnLabel
+        onChange={(e) => setValue(e.target.value)}
+        value={value}
+      />
+
+      <NumeralDate
+        label="Validation as boolean"
+        warning
+        onChange={(e) => setValue(e.target.value)}
+        value={value}
+      />
+    </>
+  );
+};
+
+export const NewValidation: ComponentStory<typeof NumeralDate> = () => {
+  const [value, setValue] = useState({ dd: "", mm: "", yyyy: "" });
+  return (
+    <CarbonProvider validationRedesignOptIn>
+      <NumeralDate
+        label="Validation as string"
+        labelHelp="Label help"
+        error="Error Message (Fix is required)"
+        onChange={(e) => setValue(e.target.value)}
+        value={value}
+      />
+
+      <NumeralDate
+        label="Validation as boolean"
+        labelHelp="Label help"
+        error
+        onChange={(e) => setValue(e.target.value)}
+        value={value}
+      />
+
+      <NumeralDate
+        label="Validation as string"
+        labelHelp="Label help"
+        warning="Warning Message (Fix is optional)"
+        onChange={(e) => setValue(e.target.value)}
+        value={value}
+      />
+
+      <NumeralDate
```

Comment:
**suggestion(non-blocking):** It might be good to put in the label what validation these are demonstrating. In this screenshot, warning doesn't have any real styling changes as a `boolean` value, so it just likes it usually does. 

![Screenshot 2023-11-24 at 09 41 55](https://github.com/Sage/carbon/assets/56251247/ae797e82-bb82-40ca-a672-cfdfbc6e13a9)